python_modules/dagster/dagster/_core/remote_representation/external_data.py

```diff
@@ -1560,213 +1552,128 @@ def external_asset_checks_from_defs(
-                is_source=execution_types_by_asset_key.get(asset_key)
-                != AssetExecutionType.MATERIALIZATION,
-                is_observable=is_observable_by_key.get(asset_key, False),
-                auto_observe_interval_minutes=auto_observe_interval_minutes_by_key.get(asset_key),
-                # assets defined by Out(asset_key="k") do not have any group
-                # name specified we default to DEFAULT_GROUP_NAME here to ensure
-                # such assets are part of the default group
-                group_name=group_name_by_asset_key.get(asset_key, DEFAULT_GROUP_NAME),
-                freshness_policy=freshness_policy_by_asset_key.get(asset_key),
-                auto_materialize_policy=auto_materialize_policy_by_asset_key.get(asset_key),
-                backfill_policy=backfill_policy_by_asset_key.get(asset_key),
-                execution_set_identifier=execution_set_identifiers.get(asset_key),
+                partitions_def_data=(
+                    external_partitions_definition_from_def(asset_node.partitions_def)
+                    if asset_node.partitions_def
+                    else None
+                ),
+                output_name=output_name,
+                metadata=asset_node.metadata,
+                tags=asset_node.tags,
+                group_name=group_name,
+                freshness_policy=asset_node.freshness_policy,
+                is_source=asset_node.is_external,
+                is_observable=asset_node.is_observable,
+                # ???
+                execution_set_identifier=execution_set_identifiers_by_asset_key.get(key),
                 required_top_level_resources=required_top_level_resources,
-                owners=[
-                    asset_owner_to_str(owner) for owner in owners_by_asset_key.get(asset_key, [])
-                ],
+                auto_materialize_policy=asset_node.auto_materialize_policy,
+                backfill_policy=asset_node.backfill_policy,
+                auto_observe_interval_minutes=asset_node.auto_observe_interval_minutes,
+                owners=asset_node.owners,
             )
         )
 
-    # Ensure any external assets that are have no nodes in any job are included in the asset graph
-    for asset in assets_defs:
-        for key in [key for key in asset.keys if key not in node_defs_by_asset_key]:
-            # This is in place to preserve an implicit behavior in the Dagster UI where stub
-            # dependencies were rendered as if they weren't part of the default asset group.
-            group_name = (
-                None
-                if asset.is_auto_created_stub
-                else group_name_by_asset_key.get(key, DEFAULT_GROUP_NAME)
-            )
-
-            asset_nodes.append(
-                ExternalAssetNode(
-                    asset_key=key,
-                    dependencies=list(deps[key].values()),
-                    depended_by=list(dep_by[key].values()),
```

Comment:
These were always empty lists because `deps` was not being populated for external assets.